bin/plugin/commands/get-plugin-version.js

```diff
@@ -38,52 +37,40 @@ function doRunGetPluginVersion( settings ) {
 		throw Error( 'A slug must be provided via the --slug (-s) argument.' );
 	}
 
-	const pluginsFile = path.join( '.', settings.pluginsJsonFile );
-
-	// Buffer contents of plugins JSON file.
-	let pluginsFileContent = '';
+	// Resolve the absolute path to the plugins.json file.
+	const pluginsFile = path.join(
+		__dirname,
+		'../../../' + settings.pluginsJsonFile
+	);
 
 	try {
-		pluginsFileContent = fs.readFileSync( pluginsFile, 'utf-8' );
-	} catch ( e ) {
-		throw Error( `Error reading file at "${ pluginsFile }": ${ e }` );
-	}
-
-	// Validate that the plugins JSON file contains content before proceeding.
-	if ( ! pluginsFileContent ) {
-		throw Error(
-			`Contents of file at "${ pluginsFile }" could not be read, or are empty.`
-		);
-	}
+		// Read the plugins.json file synchronously.
+		const { modules, plugins } = require( pluginsFile );
 
-	const pluginsConfig = JSON.parse( pluginsFileContent );
-
-	// Check for valid and not empty object resulting from plugins JSON file parse.
-	if (
-		'object' !== typeof pluginsConfig ||
-		0 === Object.keys( pluginsConfig ).length
-	) {
-		throw Error(
-			`File at "${ pluginsFile }" parsed, but detected empty/non valid JSON object.`
-		);
-	}
-
-	const plugins = pluginsConfig.modules;
-	if ( ! plugins ) {
-		throw Error(
-			`File at "${ pluginsFile }" parsed, but the modules are missing, or they are misspelled.`
-		);
-	}
+		// Validate that the modules object is not empty.
+		if ( modules && Object.keys( modules ).length !== 0 ) {
+			for ( const module of Object.values( modules ) ) {
+				if ( module.version && settings.slug === module.slug ) {
+					log( module.version );
+					return;
+				}
+			}
+		}
 
-	for ( const moduleDir in plugins ) {
-		const pluginVersion = plugins[ moduleDir ]?.version;
-		const pluginSlug = plugins[ moduleDir ]?.slug;
-		if ( pluginVersion && pluginSlug && settings.slug === pluginSlug ) {
-			return log( pluginVersion );
+		// Validate that the plugins object is not empty.
+		if ( plugins && Object.keys( plugins ).length !== 0 ) {
```

Comment:
IMHO this safeguard is overly defensive and thus overkill.

As long as there's always at the minimum `plugins: {}` in the `plugins.json` file, this is not needed. Especially the `Object.keys()` length is unnecessary, because the for loop will just do nothing on an empty array.

Thus I suggest we remove this unnecessary`if` here.